### PR TITLE
Implement rounding to 14 significant digits for on-the-fly binning

### PR DESCRIPTION
**Pull request by stefan-wallner**
When on-the-fly bins are generated from a range and a number of bins, now the bin borders are rounded to 14 significant digites to get rid of the nummeric effects of the boundary calculation. (e.g. 2.4000000000000004)

### CODE REVIEW

File: pyInterface/module/_config.py [hunk 1]
```diff
@@ -171,6 +171,11 @@ def _readBinning(binningString):
 				return []
 	return retval
```
Comment (suhlatwork) [inline pyInterface/module/_config.py]: there should be two empty lines to separate functions

File: pyInterface/module/_config.py [hunk 2]
```diff
@@ -171,6 +171,11 @@ def _readBinning(binningString):
...
 
+def _roundSignificantDigits(value, significantDigits):
+	oomValue = int( math.floor(math.log10(abs(value))))
```
Comment (suhlatwork) [inline pyInterface/module/_config.py]: there is no space to balance the one after `int(`, please write either `int( math.floor(math.log10(abs(value))) )` or `int(math.floor(math.log10(abs(value))))`

File: pyInterface/module/_config.py [hunk 3]
```diff
@@ -171,6 +171,11 @@ def _readBinning(binningString):
...
+	nDigits = significantDigits - 1 - oomValue
+	return round(value,nDigits)
```
Comment (suhlatwork) [inline pyInterface/module/_config.py]: arguments should be separated by `, `. please add the space

File: pyInterface/module/_config.py [hunk 4]
```diff
@@ -196,8 +201,12 @@ def _binningInputHandlingForCaseTwoAndThree(inputVal):
...
 				raise ValueError(errMsg)
 			expandedBoundaryList = []
-			for i in xrange(axesPartition[2]+1):
-				expandedBoundaryList.append(float(axesPartition[0]) + (float(axesPartition[1] - axesPartition[0]) / float(axesPartition[2])) * i)
+			xMin = float(axesPartition[0])
+			xMax = float(axesPartition[1])
+			nBins = int(axesPartition[2])
+			for i in xrange(nBins+1):
+				# round to 14 significant digits to get rid of numeric artifacts of the bin borders
+				expandedBoundaryList.append( _roundSignificantDigits( xMin + (xMax-xMin)/float(nBins) * i, 14) )
```
Comment (suhlatwork) [inline pyInterface/module/_config.py]: please remove the space after `_roundSignificantDigits(`

File: pyInterface/module/_config.py [hunk 5]
```diff
@@ -171,6 +171,11 @@ def _readBinning(binningString):
 				return []
 	return retval
 
+def _roundSignificantDigits(value, significantDigits):
```
Comment (suhlatwork) [inline pyInterface/module/_config.py]: IMHO `_roundToSignificantDigits`/`_roundToNSignificantDigits` would be a better name

File: pyInterface/module/_config.py (hunk 4 above)
Comment (suhlatwork) [inline pyInterface/module/_config.py]: the `float` cast of `nBins` should not change anything here, right?

**Comment by stefan-wallner** [comment]
Thanks for your comments. I agree with all of them.

**Comment by suhlatwork** [comment]
Merged, thank you!

**Pull request merged**